Simplenote/PasskeyAuthenticator.swift

```diff
@@ -3,126 +3,96 @@ import AuthenticationServices
-    }
-
-    private func fetchAuthChallenge(for email: String) async throws -> PasskeyAuthChallenge? {
-        guard let data = try await passkeyRemote.passkeyAuthChallenge(for: email) else {
-            return nil
-        }
 
-        let challenge = try JSONDecoder().decode(PasskeyAuthChallenge.self, from: data)
-        return challenge
-    }
-
-    private func performPasskeyAuthentication(with response: PasskeyAuthResponse) {
-        Task { @MainActor in
-            guard let json = try? JSONEncoder().encode(response),
-                  let response = try? await passkeyRemote.verifyPasskeyLogin(with: json),
-                  let verifyResponse = try? JSONDecoder().decode(PasskeyVerifyResponse.self, from: response) else {
-                // TODO: handle auth failure
-                return
+        return try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<PasskeyVerifyResponse, any Error>) in
+            internalAuthControllerDelegate.onCompletion = { result in
+                switch result {
+                case .success(let response):
+                    Task {
+                        do {
+                            let verify = try await self.performPasskeyAuthentication(with: response)
+                            continuation.resume(returning: verify)
+                        } catch {
+                            continuation.resume(throwing: error)
+                        }
+                    }
+
+                case .failure(let error):
+                    continuation.resume(throwing: error)
+                }
             }
 
-            authenticator.authenticate(withUsername: verifyResponse.username, token: verifyResponse.accessToken)
+            controller.performRequests()
         }
     }
-}
 
-// MARK: - ASAuthorizationControllerDelegate
-//
-extension PasskeyAuthenticator: ASAuthorizationControllerDelegate {
-    public func authorizationController(controller: ASAuthorizationController, didCompleteWithError error: any Error) {
-        // TODO: handle error
-        print(error.localizedDescription)
-    }
-
-    public func authorizationController(controller: ASAuthorizationController, didCompleteWithAuthorization authorization: ASAuthorization) {
-
-        switch authorization.credential {
-        case let credential as ASAuthorizationPlatformPublicKeyCredentialRegistration:
-            performPasskeyRegistration(with: credential)
-
-        case let credential as ASAuthorizationPlatformPublicKeyCredentialAssertion:
-            let response = PasskeyAuthResponse(from: credential)
-
-            performPasskeyAuthentication(with: response)
-        default:
-            break
+    private func performPasskeyAuthentication(with response: PasskeyAuthResponse) async throws -> PasskeyVerifyResponse {
+        guard let json = try? JSONEncoder().encode(response),
+              let response = try? await passkeyRemote.verifyPasskeyLogin(with: json) else {
```

Comment:
Quick note: **probably** the JSONEncoder step belongs in the verifyPasskeyLogin API itself.

Otherwise you could, potentially, pass any other Data!